File: user-app/lib/models/service_category.dart
class ServiceCategory {
  final int id;
  final String name;
  final String description;
  final String? iconUrl;
  final bool isActive;

  ServiceCategory({
    required this.id,
    required this.name,
    required this.description,
    this.iconUrl,
    required this.isActive,
  });

  factory ServiceCategory.fromJson(Map<String, dynamic> json) {
    return ServiceCategory(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      iconUrl: json['iconUrl'],
      isActive: json['isActive'] ?? true,
    );
  }
}

File: backend/src/main/java/com/servicebooking/model/ServiceCategory.java
package com.servicebooking.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "service_categories")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ServiceCategory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String name;

    @Column(length = 500)
    private String description;

    private String iconUrl;

    @Column(nullable = false)
    private Boolean isActive = true;
}
